templates/class-grid.html

```diff
@@ -0,0 +1,47 @@
+{% extends "auth.html" %}
+{% block regular_content %}
+<div class="flex flex-col gap-2">
+  <h2>{{_('Grid overview')}}: {{class_info.name}}</h2>
+  <div class="flex flex-row items-center justify-center mr-2 w-1/12 ml-4">
+    <select class="py-2 px-1 mt-2 h-10 text-center w-full" name="levels" id="menu">
+        {% for i in range(1, max_level + 1) %}
+            <option name="selected_level" id="select-{{i}}"
+                    value="{{ i }}" hx-trigger="change" hx-get="/grid-overview/change-level?level={{ i }}" hx-target="#level" {% if i == min_level %} selected {% endif %} >
```

Comment:
```
hx-get="/grid-overview/change-level?level={{ i }}" hx-target="#level"
```

Will replace the HTML element `<div id="level">` with the contents of fetching `http://hedywebsite/grid-overview/change-level?level=5`.

I don't see a div with `id="level"`, so there's nothing for HTMX to place the content into. This is probably why you're not seeing anything happen, but if you want to be sure open the Developer Console (`Ctrl-Shift-i` or right-click anywhere on the page and click **Inspect Element**), then check that the network request is being done and that HTMX hasn't printed errors on the JavaScript console.

It also looks like the `/grid-overview/change-level` route is rendering the `class-grid.html` template, renders a whole page (with `<html>` and `<body>` tags and everything), not just a page fragment to be dynamically loaded into a larger page.

You can do one of two things:

## 1. Refresh part of the page when the level changes

If you want a change in the dropdown to refresh only a small part of the page, use HTMX to request a Flask route that only renders a fraction of the page, namely the part that should be rendered into the `<div>` you point to. You will have:

```
/grid-overview/class/1234               -- loads the containing page and sets up the HTMX dropdown
/grid-overview/class/1234/grid/5   -- loads only the grid and nothing else for class 1234 and level 5
```

And then using HTMX:

```
/grid-overview/class/1234  --[ dynamically loads ]--> /grid-overview/class/1234/grid/5
```

## 2. Refresh the entire page when the level changes

You can also encode the level into the page load of the grid-overview page itself, and do a full page refresh to a different level when the dropdown changes.

```
/grid-overview/class/1234                 -- loads the grid for class 1234 for level=1
/grid-overview/class/1234?level=5   -- loads the grid for class 1234 for level=5
```

And then the dropdown just needs to trigger a refresh of the page.

You can do that like this:

```html
<form method="GET" action="/grid-overview/class/{{ class.id }}">
  <select name="level">
    ...
    <option value="5">Level 5</option>
    ...
  </select>

  <input type="submit" value="Change">
</form>
```

If you want the page to switch immediately as the user picks a dropdown value, so they don't have to click the submit button, you can do that with a bit of hyperscript:

```html
  <select name="level"
    _="on change "
```

Reply:
Hi @rix0rrr. Thank you so much for the tips. The first method seemed like a more convenient way and it's what I ended up using.